### ISSUE TITLE
MapExtent is not loaded correctly when opening a favourite

### ISSUE THREAD

**Issue by lexvand** [nens/lizard-client#1154]
Opening a favourite does not always set the map view extent correctly, e.g. https://utrecht.lizard.net/favourites/8e4840f9-d74a-4eb0-8822-18b19fb77f06. It sometimes shows the whole world. Not a consistent bug, so may be hard to debug.

This is supposed to be used: `"spatial": {
    "view": {
      "lat": "52.125637294945136",
      "lng": "5.092720985412597",
      "zoom": 15
    }`

**Comment by Derryrover** [comment]
I cannot open the url in description.
Created topdesk to request rights:
https://nelen-schuurmans.topdesk.net/tas/public/ssp/content/detail/incident?unid=2349fa2bb8ef4f279d673ccdc2885403

**Comment by Derryrover** [comment]
@lexvand 

Hi Lex,
I can not reproduce this bug easy (hasnot worked today yet).
This makes it incredibly hard to get anything done.
More information is better.

- What were previous succes rates of reproducing?
- What is the link to the original topdesk ticket? Or anonymized content to it if I do not have acces?
- Anything that might help

I will continue trying for now.

kr Tom

**Comment by Derryrover** [comment]
Both Hoan and Tom cannot reproduce well enough to work on this.
But I also know that Lex and Lisanne can reproduce consistently and that this is a real bug.
I put it in blocked for now

**Comment by lexvand** [comment]
I performed some additional testing with loading https://utrecht.lizard.net. Success rate was about 4 out of 30 times. The pattern I see is that it often succeeds when bootstrap takes more time to load. Most of the times the loading time of https://utrecht.lizard.net/bootstrap/lizard/ was between 80 and 280 ms. The few times that it succeeded the load time was 307, 459 and 849 ms. On the other hand there were also one or two cases of 300+ ms load times with failing rescope, but still means quite a strong pattern.

@Derryrover Is this useful information to you? Do we have to deal with internet connections or browser processing speeds that are too fast for the client?

**Comment by Derryrover** [comment]
@lexvand 
Yes this helps,
I will have another look and otherwise we can hod a in-real-life meeting one of these days.
kr Tom

**Comment by Derryrover** [comment]
Tom not able to reproduce on another 460 thinkpad (same model laptop as Lex).
Tom not able to reproduce on network Nens.
Lex can only reproduce it on his own computer (thinkpad 460).
Some clients have a similar problem.

**Comment by Derryrover** [comment]
@lexvand 

Hi Lex,

I somehow had this reproduced on dev when testing your pr.
Like to have another look at it.
Had it around 80% of the time just now.

kr Tom

**Comment by Derryrover** [comment]
Updating the global state variable sometimes seem to suffer from some race condition.
I added a $timeout as workaround.
It seems to work (as I can no longer reproduce the bug).
But since bug isnot 100% reproducable that could be coincidence

https://github.com/nens/lizard-client/pull/1161
merged without review

**Comment by lexvand** [comment]
@Derryrover 
Test results:
- In standard Chrome browser, while being logged in, the bug still occurs when going to the baseURL on staging
- After a refresh it zooms to the correct extent. This seems weird to me, as I would expect the incorrect extent that is set in the URL after a full load would be preserved. (Same happens on production, didn't notice before)
- When I zoom to a different extent and press F5 it does zoom to the previous extent as updated in the URL. This is good behavior, but deviates from the previous bullet.
- In an incognito browser the bug does hardly ever reoccur, both logged out and logged in. This is a good sign, although I can't explain the difference between normal and incognito mode.
- **Bug**: Chart favourites load in map mode.

Conclusion: Looks sort of promising. We need to fix the issue with the favourites.

**Comment by Derryrover** [comment]
@lexvand 

Hi Lex,
I have once more problems reproducing the above bullets.

- How much time would you like me to try reproducing?
- Is the fix on itself acceptable without these bullets resolved? Otherwise I might revert merging the branch and continue working in the branch.

kr Tom

**Comment by lexvand** [comment]
@Derryrover Have you fixed the chart favourite issue? If so, then I think it is worth to put it on production and see if the behavior improves there. With the bug still in it it is not acceptable.

**Comment by Derryrover** [comment]
@lexvand 
No I can also not reproduce the chart issue. On my machine on staging a chart favourite loads the chart (not the map).
https://nxt3.staging.lizard.net/favourites/5349a31c-2da3-49f7-a4a6-50dd7a4e436c

**Comment by lexvand** [comment]
It does open correctly when you open it in a new tab, not when you click it in the list of favourites (both in map and chart mode).

**Comment by Derryrover** [comment]
@lexvand 
Clicking on it within the list of favourites in the Lizard portal still worked correct for me

**Comment by lexvand** [comment]
@Derryrover I guess that just makes the minefield bigger and the behavior of any attempt to fix the issue more unpredictable. I think we should decide to leave it be.

**Comment by Derryrover** [comment]
@lexvand 

Hi Lex,

I can move this ticket back to icebox.
Should I revert the fix here?
https://github.com/nens/lizard-client/pull/1161

That is: are the bugs that you encounter caused by this fix you think?

kr Tom

**Comment by lexvand** [comment]
@Derryrover Yes, please revert. I do not feel confident about the fix given the possible bugs that result from it. We have to icebox it.

**Comment by Derryrover** [comment]
Reverted fix here:
https://github.com/nens/lizard-client/pull/1163
Left fix in comments
